modules/publish/controllers/Main.php

```diff
@@ -399,21 +399,37 @@ public function runFindArticles(framework\Request $request)
          */
         public function runToggleFavouriteArticle(framework\Request $request)
         {
-            if ($article_id = $request['article_id'])
+            // Read request parameters.
+            $article_id = $request['article_id'];
+            $user_id = $request['user_id'];
+
+            // Validate request parameters.
+            if ($article_id === null)
             {
-                try
-                {
-                    $article = Articles::getTable()->selectById($article_id);
-                    $user = \thebuggenie\core\entities\User::getB2DBTable()->selectById($request['user_id']);
-                }
-                catch (\Exception $e)
-                {
-                    return $this->renderText('fail');
-                }
+                return $this->renderText('no article');
             }
-            else
+
+            if ($user_id === null)
             {
-                return $this->renderText('no article');
+                return $this->renderText('fail');
```

Comment:
how about `return403()`?

Reply:
This one should probably be a `400`, since the request did not include correct parameter?

Reply:
Either that or a 404 since it's an item that can't be found. Given that the route requires the article_id parameter to exist to even be triggered, isn't that more likely?

Reply:
Sorry if the comments are in mixed up order, but my feedback would be 404 if the article doesn't exist and 403 if you're trying to toggle someone else's favourite.